autoscale/src/main/java/com/sequenceiq/periscope/service/AltusMachineUserService.java

```diff
@@ -97,4 +115,13 @@ private void syncEnvironment(String accountId, String machineUserCrn, String env
         LOGGER.info("Finished invoking freeIpa user Sync with operation: {}, for environmentCrn: {} and machineUserCrn: {}", statusResponse.getOperationId(),
                 environmentCrn, machineUserCrn);
     }
+
+    private boolean userSyncNeeded(String stackCrn) {
+        String cbVersion = cloudbreakVersionService.getCloudbreakSaltStateVersionByStackCrn(stackCrn);
+        if (!Strings.isNullOrEmpty(cbVersion)) {
+            String shortened = StringUtils.substringBefore(cbVersion, "-");
+            return VERSION_COMPARATOR.compare(() -> shortened, () -> cloudbreakVersionThreshold) < 0;
+        }
+        return true;
+    }
```

Comment:
What is the rationale for basing the decision on the CB version? Does the machine user automagically exist in CB 2.70 and does not to be synced to the cluster?

Reply:
#14331 went into 2.70. That patch includes explicitly whitelisting the autoscale machineUser for new DH clusters. We're making a check against that version to determine if userSync was needed or not